recipe/meta.yaml

```diff
@@ -25,8 +25,8 @@ requirements:
     - meson-python ==0.13.1
     - meson ==1.2.1
     - cython >=3.0.5,<4
-    - numpy 1.22  # [py<311]
-    - numpy {{ numpy }}  # [py>=311]
+    - numpy >=1.22,<2.0  # [py<310]
+    - numpy >=2.0,<2.3  # [py>=310]
     - versioneer
     - pip
   run:
```

Comment:
Can you adjust the numpy depenedency in the run section?
When built with numpy 2, pin_compatible is too restrictive.